src/Illuminate/Contracts/Routing/UrlRoutable.php

```diff
@@ -17,4 +17,11 @@ public function getRouteKey();
      * @return string
      */
     public function getRouteKeyName();
+
+    /**
+     * Get the qualified route key for the model.
+     *
+     * @return string
+     */
+    public function getRouteQualifiedKeyName();
```

Comment:
I though UrlRoutable interface is being implemented and used in Eloquent\Model class only (atleast what I see after searching the codebase) or is there any other use cases? If yes, then it would probably make sense to add a ModelRoutable (or some) Contract and move getRouteQualifiedKeyName there.

Please advice!